checkers/python/avoid-mutable-default-arg.yml

```diff
@@ -0,0 +1,24 @@
+language: py
+name: avoid-mutable-default-arg
+message: Avoid using mutable default arguments in function definitions. Use None as the default and initialize the mutable object inside the function.
+category: bug-risk
+severity: error
+
+pattern: |
+  (function_definition
+    parameters: (parameters
+      (default_parameter
+        name: (identifier) @param
+        value: (list | dict | set) @mutable)))
```

Comment:
The pattern appears to be invalid. Please look at the CI logs. Ideally, you should run `make test-builtin-rules` locally before creating a PR. Make sure you run `make build` before running the tests.